smriprep/workflows/anatomical.py

```diff
@@ -27,17 +28,194 @@
+            bspline_fitting_distance=200),
+        n_procs=omp_nthreads, name='inu_n4_final', iterfield=['input_image'])
+
+    # Tolerate missing ANTs at construction time
+    _ants_version = N4BiasFieldCorrection().version
+
+    if _ants_version and parseversion(_ants_version) >= Version('2.1.0'):
+        inu_n4_final.inputs.rescale_intensities = True
+    else:
+        warn("""\
+Found ANTs version %s, which is too old. Please consider upgrading to 2.1.0 or \
+greater so that the --rescale-intensities option is available with \
+N4BiasFieldCorrection.""" % _ants_version, DeprecationWarning)
+
+    copy_xform = pe.Node(CopyXForm(
+        fields=['out_file', 'bias_image']),
+        name='copy_xform', run_without_submitting=True)
+
+    wf.connect([
+        (inputnode, inu_n4_final, [('in_files', 'input_image')]),
+        (inputnode, thr_brainmask, [(('in_files', _pop), 'in_file')]),
+        (thr_brainmask, outputnode, [('out_mask', 'out_mask')]),
+        (inputnode, copy_xform, [(('in_files', _pop), 'hdr_file')]),
+        (inu_n4_final, copy_xform, [('output_image', 'out_file')]),
+        (inu_n4_final, copy_xform, [('bias_image', 'bias_image')]),
+        (copy_xform, outputnode, [('out_file', 'out_file')]),
+        (copy_xform, outputnode, [('out_file', 'bias_corrected')]),
+        (copy_xform, outputnode, [('bias_image', 'bias_image')])
+    ])
+
+    # If atropos refine, do in4 twice
+    if atropos_refine:
+        atropos_model = atropos_model or list(
+            ATROPOS_MODELS[bids_suffix].values())
+        atropos_wf = init_atropos_wf(
+            use_random_seed=atropos_use_random_seed,
+            omp_nthreads=omp_nthreads,
+            mem_gb=mem_gb,
+            in_segmentation_model=atropos_model,
+        )
+        sel_wm = pe.Node(niu.Select(index=atropos_model[-1] - 1), name='sel_wm',
+                         run_without_submitting=True)
+
+        inu_n4 = pe.MapNode(
+            N4BiasFieldCorrection(
+                dimension=3, save_bias=False, copy_header=True,
+                n_iterations=[50] * 4, convergence_threshold=1e-7,
+                shrink_factor=4, bspline_fitting_distance=200),
+            n_procs=omp_nthreads, name='inu_n4', iterfield=['input_image'])
+
+        wf.connect([
+            (inputnode, inu_n4, [('in_files', 'input_image')]),
+            (inu_n4, atropos_wf, [
+                ('output_image', 'inputnode.in_files')]),
+            (thr_brainmask, atropos_wf, [
+                ('out_mask', 'inputnode.in_mask')]),
+            (thr_brainmask, atropos_wf, [
+                ('out_mask', 'inputnode.in_mask_dilated')]),  # Dilate?
+            (atropos_wf, sel_wm, [('outputnode.out_tpms', 'inlist')]),
+            (sel_wm, inu_n4_final, [('out', 'weight_image')]),
+            (atropos_wf, outputnode, [
+                ('outputnode.out_segm', 'out_segm'),
+                ('outputnode.out_tpms', 'out_tpms')]),
+        ])
```

Comment:
I'm not going to ask you to do that in this PR, but I think this could be better modularized in niworkflows: https://github.com/poldracklab/niworkflows/issues/424

Reply:
Yes, agreed